### PR TITLE
✨ add request revisions modal

### PR DESCRIPTION
**Pull request by samrichca**
- add global modal component, cribbed from platform-ui
- add new yellow header to applications entity page, with request revisions button
- add request revisions modal
- set up data for data hookup ticket

### CODE REVIEW

File: components/Root.tsx [hunk 1]
```diff
@@ -1,9 +1,69 @@
...
+import { css } from '@emotion/core';
+import ReactDOM from 'react-dom';
+
 import ThemeProvider from '@icgc-argo/uikit/ThemeProvider';
+import Modal from '@icgc-argo/uikit/Modal';
 
 import Head from 'components/Head';
 import { AuthProvider } from 'global/hooks/useAuthContext';
 
+/**
```
Comment (samrichca) [inline components/Root.tsx:11]: from platform-ui

File: components/pages/Applications/ApplicationForm/RequestRevisionsBar/RequestRevisionsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          aria-label={`${title} textarea`}
+          id={`${title}-textarea`}
+          css={css`
+          height: 69px;
+          margin-bottom: 0;
+          width: 550px;
+        `}
+          value={details}
+          onChange={(e) => dispatch({ payload: e.target.value, type: 'details', ...dispatchArgs })}
+          onBlur={() => {
```
Comment (samrichca) [inline components/pages/Applications/ApplicationForm/RequestRevisionsBar/RequestRevisionsModal.tsx:94]: onBlur & onChange might change in data hookup ticket

File: components/pages/Applications/ApplicationForm/RequestRevisionsBar/types.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export type RequestRevisionsFieldNames = keyof typeof RequestRevisionsFieldTitles;
+
+export type RequestRevisionProperties = {
+  details: string;
+  requested: boolean;
+};
+
+export type RequestRevisionsFieldsState = {
+  [key in RequestRevisionsFieldNames]: RequestRevisionProperties;
```
Comment (samrichca) [inline components/pages/Applications/ApplicationForm/RequestRevisionsBar/types.ts:19]: i based this structure on the API request body needed in the next ticket

File: components/Root.tsx [hunk 4]
```diff
@@ -1,9 +1,69 @@
...
+
+const modalPortalRef = React.createRef<HTMLDivElement>();
+const useMounted = () => {
+  const [mounted, setMounted] = React.useState(false);
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+  return mounted;
+};
+export const ModalPortal = ({ children }: { children: React.ReactElement }) => {
```
Comment (caravinci) [inline components/Root.tsx:37]: Think we could isolate all these `const` into a separate HOC... like a "portal provider".
Thoughts?

**Review by caravinci — APPROVED** [review]
Aside from a TODO (making the a portal provider), looks good to me without testing it locally.

**Pull request merged**